src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+public class BookDaoImpl implements BookDao {
+    private static final String ID = "id";
+    private static final String TITLE = "title";
+    private static final String PRICE = "price";
+    private static final String EXPECTED_MESSAGE =
+            "Expected to insert at leas one row, but inserted 0 rows";
+    private static final String DB_CONNECTION_ERROR_MESSAGE =
+            "Can't create a connection to the DB";
+    private static final String UPDATE_ERROR_MESSAGE = "Can't update book by id: ";
+    private static final String DELETE_ERROR_MESSAGE = "Can't delete book by id: ";
+    private static final String NULL_ID_ERROR_MESSAGE = "Id can't be null";
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException(EXPECTED_MESSAGE);
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(DB_CONNECTION_ERROR_MESSAGE, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
```

Comment:
Your query is here in method, but parameters (like TITLE) are constants. Be consistent in this thing, so whether use constants both for query and params or local/inline in a method. (more preferred to use constants, but in real life it would depend on a specific project conventions)